src/query_parser/v2/parser.y

```diff
@@ -733,67 +684,91 @@ text_expr(A) ::= PERCENT PERCENT PERCENT param_term(B) PERCENT PERCENT PERCENT.
 
 modifier(A) ::= MODIFIER(B) . {
   B.len = unescapen((char*)B.s, B.len);
-  A = B;
+  A.tok = B;
+  if (ctx->sctx->spec) {
+    A.fs = IndexSpec_GetFieldWithLength(ctx->sctx->spec, B.s, B.len);
+    if (!A.fs) {
+      reportSyntaxError(ctx->status, &A.tok, "Unknown field");
+    }
+  }
 }
 
 modifierlist(A) ::= modifier(B) OR term(C). {
-  if (__builtin_expect(C.len > 0, 1)) {
-    A = NewVector(char *, 2);
-    char *s = rm_strndup(B.s, B.len);
-    Vector_Push(A, s);
-    s = rm_strndup(C.s, C.len);
-    Vector_Push(A, s);
+  if (ctx->sctx->spec) {
+    if (!FIELD_IS(B.fs, INDEXFLD_T_FULLTEXT)) {
+      REPORT_WRONG_FIELD_TYPE(B, SPEC_TEXT_STR);
+      A = NULL;
+    } else {
```

Comment:
We couldn't really search on tags with this modifier list before right?
If so, great! (breaking, but great!)

Reply:
I deduced from the rule (singular) that uses the `modifierlist` type that we only allow it for `TEXT` fields. Were you able to run a `@t1|t2:{tag}` query with dialect > 1?